phalcon/acl/adapter/memory.zep

```diff
@@ -780,6 +749,88 @@ class Memory extends Adapter
 		return haveAccess == Acl::ALLOW;
 	}
 
+	/**
+	 * Check whether a role is allowed to access an action from a resource
+	 */
+	protected function _isAllowed(string roleName, string resourceName, string access) -> string | boolean
+    {
+        var accessList, accessKey,checkRoleToInherit, checkRoleToInherits, usedRoleToInherits, usedRoleToInherit;
+
+		let accessList = this->_access;
+
+        let accessKey = roleName . "!" . resourceName . "!" . access;
+
+		/**
+		 * Check if there is a direct combination for role-resource-access
+		 */
+		if isset accessList[accessKey] {
+			return accessKey;
+		}
+		/**
+         * Check if there is a direct combination for role-*-*
+         */
+        let accessKey = roleName . "!" . resourceName . "!*";
+        if isset accessList[accessKey] {
+            return accessKey;
+        }
+        /**
+         * Check if there is a direct combination for role-*-*
+         */
+        let accessKey = roleName . "!*!*";
+        if isset accessList[accessKey] {
+            return accessKey;
+        }
+        /**
+         * Deep check if the role to inherit is valid
+         */
+        if isset this->_roleInherits[roleName] {
+            let checkRoleToInherits = [];
+            for usedRoleToInherit in this->_roleInherits[roleName] {
+                array_push(checkRoleToInherits,usedRoleToInherit);
```

Comment:
Use `let checkRoleToInherits[] =`

Reply:
let checkRoleToInherits[] = in php5.x is error
when array_​shift checkRoleToInherits the this->_roleInherits[roleName] is delete too。
look commit 168f51c

Reply:
But you don't use there [], you use there +, for sure we add in other places lements to array like `let array[] = `

Reply:
It is ok now?